lib/bunch.ex

```diff
@@ -24,6 +24,18 @@ defmodule Bunch do
 
   @compile {:inline, listify: 1, error_if_nil: 2}
 
+  @doc """
+  Extracts the key from a key-value tuple.
+  """
+  @spec key({key, value}) :: key when key: any, value: any
+  def key({key, _value}), do: key
+
+  @doc """
+  Extracts the value from a key-value tuple.
+  """
+  @spec value({key, value}) :: value when key: any, value: any
+  def value({_key, value}), do: value
+
```

Comment:
Honestly I wonder it these two could maybe be called `fst` and `snd` like in Haskell: https://hackage.haskell.org/package/base-4.12.0.0/docs/Prelude.html#v:fst

These names have much broader meaning